packages/uploadthing/src/internal/express.ts

```diff
@@ -0,0 +1,68 @@
+import { Router as ExpressRouter } from "express";
+
+import { getStatusCodeFromError, UploadThingError } from "@uploadthing/shared";
+
+import { UPLOADTHING_VERSION } from "../constants";
+import { defaultErrorFormatter } from "./error-formatter";
+import type { RouterWithConfig } from "./handler";
+import { buildPermissionsInfoHandler, buildRequestHandler } from "./handler";
+import type { FileRouter, inferErrorShape } from "./types";
+
+export const createExpressRouter = <TRouter extends FileRouter>(
+  opts: RouterWithConfig<TRouter>,
+): ExpressRouter => {
+  const requestHandler = buildRequestHandler<TRouter, "express">(opts);
+  const router = ExpressRouter();
+
+  // eslint-disable-next-line @typescript-eslint/no-misused-promises
+  router.post("/", async (req, res) => {
+    const response = await requestHandler({
+      req: {
+        ...req,
+        // Explicitly passing headers since
+        // without that they are being lost for some reason
+        headers: req.headers,
+        json: () => new Promise((resolve) => resolve(req.body)),
+      },
+      res,
+    });
+    const errorFormatter =
+      opts.router[Object.keys(opts.router)[0]]?._def.errorFormatter ??
+      defaultErrorFormatter;
+
+    if (response instanceof UploadThingError) {
+      const formattedError = errorFormatter(
+        response,
+      ) as inferErrorShape<TRouter>;
+
+      res.status(getStatusCodeFromError(response));
+      res.setHeader("x-uploadthing-version", UPLOADTHING_VERSION);
+      res.send(JSON.stringify(formattedError));
+
+      return;
+    }
+
+    if (response.status !== 200) {
+      // We messed up - this should never happen
+      res.status(500);
+      res.setHeader("x-uploadthing-version", UPLOADTHING_VERSION);
+      res.send("An unknown error occured");
+
+      return;
+    }
+
+    res.status(response.status);
+    res.setHeader("x-uploadthing-version", UPLOADTHING_VERSION);
+    res.send(JSON.stringify(response.body));
+  });
+
+  const getBuildPerms = buildPermissionsInfoHandler<TRouter>(opts);
+
+  router.get("/", (req, res) => {
+    res.status(200);
+    res.setHeader("x-uploadthing-version", UPLOADTHING_VERSION);
+    res.send(JSON.stringify(getBuildPerms()));
+  });
+
+  return router;
```

Comment:
Since we require the api to be on a specific `/api/uploadthing` endpoint, could we return that here, and bypass the user needing to add it when they register the middleware?

Reply:
Does not look like so. Btw, TRPC also require user to specify the endpoint
<img width="419" alt="image" src="https://github.com/pingdotgg/uploadthing/assets/55754540/7c69fabd-b1fc-4069-8804-969c5d28ee95">

I checked Express docs and could not find anything useful to address that

Reply:
Fair point, I guess there is precedent for requiring the use to enter a specific endpoint. 

It's just a function, so you should be able to return an array, then spread it into the app.use(), but that would probably be most confusing to most users who would be using this.

Reply:
> Btw, TRPC also require user to specify the endpoint

tRPC doesn't require you to serve your router on that endpoint though

> It's just a function, so you should be able to return an array, then spread it into the `app.use()`, but that would probably be most confusing to most users who would be using this.

That's what i was thinking. Not sure if it's better or worse UX

Reply:
I thought about such option but decided to not suggest it as it looks too hacky to me.

Btw, are there any plans on allowing serving UT API on other endpoints? If there is such thing in mind, I would recommend current approach